numba/core/typing/builtins.py

```diff
@@ -804,6 +804,15 @@ def typer(val):
             elif isinstance(val, (types.Number, types.Boolean)):
                  # Scalar constructor, e.g. np.int32(42)
                  return ty
+            elif isinstance(val, (types.NPDatetime, types.NPTimedelta)):
+                # Constructor cast from datetime-like, e.g.
+                # > np.int64(np.datetime64("2000-01-01"))
+                if ty.bitwidth == 64:
+                    return ty
+                else:
+                    msg = (f"Cannot cast {val} to {ty} as {ty} is not 64 bits "
+                           "wide.")
+                    raise errors.TypingError(msg)
```

Comment:
@stuartarchibald, NumPy allows casting datatypes to other integer types. Should Numba follow the same convention?

```python
from numba import njit
import numpy as np

@njit
def foo(a):
    return np.int8(a)

a = np.timedelta64('NaT')
print(foo.py_func(a))
```